src/main/java/seedu/address/model/person/Gender.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's gender in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidGender(String)}
+ */
+public class Gender {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Gender should only contain alphabets, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alpha} ]*";
+
+    public final String value;
+
+    /**
+     * Constructs a {@code Gender}.
+     *
+     * @param gender A valid gender.
+     */
+    public Gender(String gender) {
+        requireNonNull(gender);
+        checkArgument(isValidGender(gender), MESSAGE_CONSTRAINTS);
+        this.value = gender;
+    }
+
+    /**
+     * Returns true if a given string is a valid gender.
+     */
+    public static boolean isValidGender(String test) {
+        if (test.matches(VALIDATION_REGEX)) {
+            if (test.equalsIgnoreCase("male") || test.equalsIgnoreCase("female")) {
```

Comment:
this implementation is ok. But if you want to avoid nested ifs altogether, you can do if (!test.matches(...)) {return false;} if (! test.....) {return false;} return true; But either way it's ok.

Reply:
ok thank you!